mypy/plugin.py

```diff
@@ -379,3 +407,289 @@ def int_pow_callback(ctx: MethodContext) -> Type:
         else:
             return ctx.api.named_generic_type('builtins.float', [])
     return ctx.default_return_type
+
+
+# Arguments to the attr functions (attr.s and attr.ib) with their defaults.
+ArgumentInfo = NamedTuple(
+    'ArgumentInfo', [
+        ('default', Optional[bool]),
+        ('kwarg_name', Optional[str]),
+        ('index', Optional[int]),
+    ])
+
+AttrClassMaker = NamedTuple(
+    'AttrClassMaker', [
+        ('init', ArgumentInfo),
+        ('frozen', ArgumentInfo),
+        ('cmp', ArgumentInfo),
+        ('auto_attribs', ArgumentInfo)
+    ]
+)
+
+AttribMaker = NamedTuple(
+    'AttribMaker', [
+        ('default', ArgumentInfo),
+        ('init', ArgumentInfo),
+        ("type", ArgumentInfo),
+    ]
+)
+
+attrs_arguments = AttrClassMaker(
+    cmp=ArgumentInfo(True, 'cmp', 4),
+    init=ArgumentInfo(True, 'init', 6),
+    frozen=ArgumentInfo(False, 'frozen', 8),
+    auto_attribs=ArgumentInfo(False, 'auto_attribs', 10)
+)
+attrib_arguments = AttribMaker(
+    default=ArgumentInfo(None, 'default', 0),
+    init=ArgumentInfo(True, "init", 5),
+    type=ArgumentInfo(None, 'type', 8),
+)
+
+
+# The names of the different functions that create classes or arguments.
+attr_class_makers = {
```

Comment:
I could use some advice/opinion here. It's fairly common when using attrs to write your own wrappers around the functions to make coding easier.  However the plugins only work on specific names so I tried to leave some way for users to add their own "class makers" and "attrib makers" to the plugin.  But that's probably not the right way to do this.